Sources/BraintreeCore/Authorization/TokenizationKey.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+class TokenizationKey: ClientAuthorization {
+    
+    // MARK: - Internal Properties
+    
+    let type = AuthorizationType.tokenizationKey
+    let bearer: String
+    let configURL: URL
+    let originalValue: String
+    
+    // MARK: - Initializer
+    
+    init(_ rawValue: String) throws {
+        self.bearer = rawValue
+        self.originalValue = rawValue
+        guard let configURL = TokenizationKey.baseURLFromTokenizationKey(rawValue) else {
+            throw TokenizationKeyError.invalid
+        }
+        self.configURL = configURL
+    }
+    
+    // MARK: - Private Methods
+    
+    private static func baseURLFromTokenizationKey(_ tokenizationKey: String) -> URL? {
+        let pattern: String = "([a-zA-Z0-9]+)_[a-zA-Z0-9]+_([a-zA-Z0-9_]+)"
+        guard let regularExpression = try? NSRegularExpression(pattern: pattern) else { return nil }
+
+        let range = NSRange(location: 0, length: tokenizationKey.count)
+        let matches = regularExpression.matches(in: tokenizationKey, range: range)
+
+        if matches.count != 1 || matches.first?.numberOfRanges != 3 {
+            return nil
+        }
+
+        var environment: String = ""
+        var merchantID: String = ""
+
+        matches.forEach { match in
+            environment = (tokenizationKey as NSString).substring(with: match.range(at: 1))
+            merchantID = (tokenizationKey as NSString).substring(with: match.range(at: 2))
+        }
```

Comment:
If we wanted to match the Android implementation here we can simplify this. It looks like they do the [regex check here](https://github.com/braintree/braintree_android/blob/main/BraintreeCore/src/main/java/com/braintreepayments/api/Authorization.kt#L58) then use [split here](https://github.com/braintree/braintree_android/blob/95758a0b5fece9131f7386209492134c7525bb1f/BraintreeCore/src/main/java/com/braintreepayments/api/TokenizationKey.kt#L19-L21). Would be something like:
```swift
        let pattern: String = "([a-zA-Z0-9]+)_[a-zA-Z0-9]+_([a-zA-Z0-9_]+)"
        guard tokenizationKey.range(of: pattern, options: .regularExpression) != nil else { return nil }

        let tokenizationKeyParts = tokenizationKey.split(separator: "_", maxSplits: 3)
        let environment: String = String(tokenizationKeyParts[0])
        let merchantID: String = String(tokenizationKeyParts[2])
```

Reply:
Awesome! Applied in c88ec7dab0f18760a611c34d1fd18d22b89293f8